packages/components/src/List/ListComposition/SortBy/SortBy.component.js

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Navbar, NavDropdown, Nav, NavItem, MenuItem } from 'react-bootstrap';
+import uuid from 'uuid';
+import isUndefined from 'lodash/isUndefined';
+
+import { useListContext } from '../context';
+
+/**
+ * Get the component's selected sort by option
+ * @param {object} props Props passed to the component
+ * @param {object} sortParams Current value of the context's `sortParams`
+ * @returns {string}
+ */
+function getSelectedOption(props, sortParams) {
+	if (props.selected) {
+		return props.selected; // From props (controlled mode)
+	} else if (sortParams.sortBy) {
+		return sortParams.sortBy; // From context (uncontrolled)
+	}
+	return props.options[0].key; // Default to first available option
```

Comment:
This is not correct.
The state is held either by props, either by the context. If none has a value, it means that there is no sort. In old list we have `N.C` or something like that.

Knowing that, I think you can
* remove this function, and just use `props.selected || sortParams.sortBy`.
* remove the useEffect